modules/sdk-coin-dot/src/lib/utils.ts

```diff
@@ -264,6 +279,139 @@ export class Utils implements BaseUtils {
     return (arg as TransferAllArgs).dest?.id !== undefined && (arg as TransferAllArgs).keepAlive !== undefined;
   }
 
+  /**
+   * Returns true if arg is of type BatchArgs, false otherwise.
+   *
+   * @param arg The object to test.
+   *
+   * @return true if arg is of type BatchArgs, false otherwise.
+   */
+  isBatch(arg: TxMethod['args']): arg is BatchArgs {
+    return (arg as BatchArgs).calls !== undefined;
+  }
+
+  /**
+   * Returns true if arg is of type BatchArgs and the calls of the batch are staking calls: a stake
+   * call (bond) followed by an add proxy call (addProxy), false otherwise.
+   *
+   * @param arg The object to test.
+   *
+   * @return true if arg is of type BatchArgs and the calls of the batch are staking calls: a stake
+   * call (bond) followed by an add proxy call (addProxy), false otherwise.
+   */
+  isStakingBatch(arg: TxMethod['args']): arg is BatchArgs {
+    const calls = (arg as BatchArgs).calls;
+    if (calls !== undefined) {
+      return (
+        calls.length === 2 && this.isStakeBatchCallArgs(calls[0].args) && this.isAddProxyBatchCallArgs(calls[1].args)
+      );
+    }
+    return false;
+  }
+
+  /**
+   * Returns true if arg is of type StakeBatchCallArgs, false otherwise.
+   *
+   * @param arg The object to test.
+   *
+   * @return true if arg is of type StakeBatchCallArgs, false otherwise.
+   */
+  isStakeBatchCallArgs(arg: BatchCallObject['args']): arg is StakeBatchCallArgs {
+    return (
+      (arg as StakeBatchCallArgs).value !== undefined &&
+      (arg as StakeBatchCallArgs).controller !== undefined &&
+      (arg as StakeBatchCallArgs).payee !== undefined
+    );
+  }
+
+  /**
+   * Returns true if arg is of type AddProxyBatchCallArgs, false otherwise.
+   *
+   * @param arg The object to test.
+   *
+   * @return true if arg is of type AddProxyBatchCallArgs, false otherwise.
+   */
+  isAddProxyBatchCallArgs(arg: BatchCallObject['args']): arg is AddProxyBatchCallArgs {
+    return (
+      (arg as AddProxyBatchCallArgs).delegate !== undefined &&
+      (arg as AddProxyBatchCallArgs).proxy_type !== undefined &&
+      (arg as AddProxyBatchCallArgs).delay !== undefined
+    );
+  }
+
+  /**
+   * Returns true if arg is of type BatchArgs and the calls of the batch are unstaking calls: a remove
+   * proxy call (removeProxy), followed by a chill call, and an unstake call (unbond), false otherwise.
+   *
+   * @param arg The object to test.
+   *
+   * @return true if arg is of type BatchArgs and the calls of the batch are unstaking calls: a remove
+   * proxy call (removeProxy), followed by a chill call, and an unstake call (unbond), false otherwise.
+   */
+  isUnstakingBatch(arg: TxMethod['args']): arg is BatchArgs {
+    const calls = (arg as BatchArgs).calls;
+    if (calls !== undefined) {
+      return (
+        calls.length === 3 &&
+        this.isRemoveProxyBatchCallArgs(calls[0].args) &&
```

Comment:
what is a chill call, do we not need to verify it?

Reply:
Good point, chill has no arguments so it's an empty, added a check for it.